firedrake/matrix.py

```diff
@@ -146,6 +151,7 @@ class ImplicitMatrix(MatrixBase):
         :class:`Matrix`.  May be `None` if there are no boundary
         conditions to apply.
 
+    :kwarg fc_params: a dict of form compiler parameters for this matrix.
```

Comment:
```suggestion
    :kwarg form_compiler_parameters: a dict of form compiler parameters for this matrix.
```

Reply:
This is going to break API

Reply:
On that note, I know you haven't changed `options_prefix` or `appctx` kwargs here, but could you add them to the docstring? Currently you just have to _know_ that you can pass them.